tests/sharedv4/sharedv4_test.go

```diff
@@ -0,0 +1,269 @@
+		nodesToKill := []node.Node{}
+		var err error
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("nfskill-%d", i))...)
+		}
+
+		ValidateApplications(contexts)
+		for _, ctx := range contexts {
+			var appVolumes []*volume.Volume
+			Step(fmt.Sprintf("get volumes for %s app", ctx.App.Key), func() {
+				appVolumes, err = Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(appVolumes).NotTo(BeEmpty())
+			})
+			for _, v := range appVolumes {
+
+				Step("get attached node and stop the instance", func() {
+					currNodes := node.GetStorageDriverNodes()
+					countOfCurrNodes := len(currNodes)
+
+					attachedNode, err := Inst().V.GetNodeForVolume(v, defaultCommandTimeout, defaultCommandRetry)
+					nodesToKill = append(nodesToKill, *attachedNode)
+
+					// Reboot node and check driver status
+					Step(fmt.Sprintf("stop node : %v having volume: %v attached", attachedNode.Name, v.Name), func() {
+
+						KillANodeAndValidate(nodesToKill)
+
+						Step("validate applications", func() {
+							for _, ctx := range contexts {
+								ValidateContext(ctx)
+							}
+						})
+
+						Step(fmt.Sprintf("validate node: %v is stopped", attachedNode.Name), func() {
+							currNodes = node.GetStorageDriverNodes()
+							isNodePresent := false
+							for _, currNode := range currNodes {
+								if currNode.Name == attachedNode.Name {
+									isNodePresent = true
+									break
+								}
+							}
+							Expect(isNodePresent).To(BeFalse())
+						})
+
+						Step(fmt.Sprintf("wait to new instance to start scheduler: %s and volume driver: %s",
+							Inst().S.String(), Inst().V.String()), func() {
+							time.Sleep(2 * time.Minute)
+							currNodes = node.GetStorageDriverNodes()
+							Expect(countOfCurrNodes).To(Equal(len(currNodes)))
+							for _, n := range currNodes {
+
+								err = Inst().S.IsNodeReady(n)
+								Expect(err).NotTo(HaveOccurred())
+
+								err = Inst().V.WaitDriverUpOnNode(n, Inst().DriverStartTimeout)
+								Expect(err).NotTo(HaveOccurred())
+							}
+						})
+
+						Step("validate apps", func() {
+							for _, ctx := range contexts {
+								ValidateContext(ctx)
+							}
+						})
+
+					})
+				})
+			}
+
+		}
+
+	})
+	JustAfterEach(func() {
+		AfterEachTest(contexts)
+	})
+})
+
+func TestMain(m *testing.M) {
+	// call flag.Parse() here if TestMain uses flags
+	ParseFlags()
+	os.Exit(m.Run())
+}
+
+func KillANodeAndValidate(storageDriverNodes []node.Node) {
+	rand.Seed(time.Now().Unix())
+	nodeToKill := storageDriverNodes[rand.Intn(len(storageDriverNodes))]
+
+	Step(fmt.Sprintf("Deleting node [%v]", nodeToKill.Name), func() {
+		logrus.Infof("Instance is of %v ", Inst().N.String())
+		err := Inst().N.DeleteNode(nodeToKill, nodeDeleteTimeoutMins)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Step("Wait for 10 min. to node get replaced by autoscalling group", func() {
+		time.Sleep(10 * time.Minute)
+	})
```

Comment:
Similar question to the above. Hardcoded sleep interval may be prone to intermittent failures or unnecessarily long sleep. Ideally, use a loop with short sleep interval and long timeout.

Reply:
Update the logic to check and wait